releases/release-5.0.4.md

```diff
@@ -0,0 +1,165 @@
+
+## Bug Fixes
+
++ TiDB
+
+    - tidb panic while query hash partition table with is null condition [#26963](https://github.com/pingcap/tidb/pull/26963)
+    - planner: fix the unstable test case TestAnalyzeIncremental [#26851](https://github.com/pingcap/tidb/pull/26851)
+    - Keep the overflow check logic same with mysql [#26724](https://github.com/pingcap/tidb/pull/26724)
+    - Fix wrong charset and collation for case when function [#26672](https://github.com/pingcap/tidb/pull/26672)
+    - Fix the case that TiDB would return the wrong result when the children of the UNION contain pure NULL values [#26571](https://github.com/pingcap/tidb/pull/26571)
+    - Fix the issue that greatest(datetime) union null returns empty string [#26565](https://github.com/pingcap/tidb/pull/26565)
+    - fix incompatible last_day func behavior in sql mode [#26000](https://github.com/pingcap/tidb/pull/26000)
+    - Fix the issue that committing pessimistic transactions may report write-conflict errors. [#25974](https://github.com/pingcap/tidb/pull/25974)
+    - Fix the wrong result after "insert ignore on duplicate update" in the secondary index has the same column in primary key but has prefix length [#25905](https://github.com/pingcap/tidb/pull/25905)
+    - Fix the wrong result for "insert ignore duplicate up" on partition table when handle changed and duplicated on secondary index [#25859](https://github.com/pingcap/tidb/pull/25859)
+    - fix wrong enum key in point get [#24772](https://github.com/pingcap/tidb/pull/24772)
+    - fix dml_batch_size doesn't load the global variable [#24731](https://github.com/pingcap/tidb/pull/24731)
+    - planner: filter conflict read_from_storage hints [#24374](https://github.com/pingcap/tidb/pull/24374)
+    - fix wrong type infer for agg function when type is null. [#24354](https://github.com/pingcap/tidb/pull/24354)
+    - fix wrong flen infer for bit constant [#24267](https://github.com/pingcap/tidb/pull/24267)
+    - fix some potential panic in statistics [#24061](https://github.com/pingcap/tidb/pull/24061)
+    - fix approx_percent panic on bit column [#23703](https://github.com/pingcap/tidb/pull/23703)
+    - fix get var expr when session var is hex literal [#23373](https://github.com/pingcap/tidb/pull/23373)
+    - fix unexpected constant fold when year compare string. [#23336](https://github.com/pingcap/tidb/pull/23336)
+
++ PD
+
+    - Fix the bug that PD would not fix down-peer in time. [#4082](https://github.com/tikv/pd/pull/4082)
+    - Fix an issue where data is not stored when using max-replicas or location-labels to indirectly update default placement rule [#3914](https://github.com/tikv/pd/pull/3914)
+    - Fix the bug that PD may panic during scaling out TiKV. [#3910](https://github.com/tikv/pd/pull/3910)
+
+## 以下 note 未分类。请将以下 note 进行分类 (Feature enhancements, Improvements, Bug fixes, Compatibility Changes 四类)，并移动到上面对应的标题下。如果某条 note 为多余的，请删除。如果漏抓取了 note，请手动补充
+
++ TiDB
```

Comment:
Note: these release notes are removed because they are unnecessary:
- Fix an issue that the `TABLESAMPLE` query result from partitioned tables is not sorted as expected. [#27412](https://github.com/pingcap/tidb/pull/27412)
- The undocumented `/debug/sub-optimal-plan` HTTP API has been removed. [#27263](https://github.com/pingcap/tidb/pull/27263)
- planner: fix the unstable test case TestAnalyzeIncremental [#26851](https://github.com/pingcap/tidb/pull/26851)